src/main/java/spot/Parser.java

```diff
@@ -0,0 +1,134 @@
+package spot;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+
+import spot.command.AddDeadlineCommand;
+import spot.command.AddEventCommand;
+import spot.command.AddToDoCommand;
+import spot.command.Command;
+import spot.command.DeleteCommand;
+import spot.command.ExitCommand;
+import spot.command.FindCommand;
+import spot.command.ListCommand;
+import spot.command.ListTasksOnCommand;
+import spot.command.MarkCommand;
+import spot.command.UnmarkCommand;
+import spot.exception.SpotException;
+
+
+/**
+ * Represents a parser to parse commands and dates.
+ */
+public class Parser {
+
+    /**
+     * Returns command corresponding to the specified input string.
+     *
+     * @param input Input string to be parsed.
+     * @return Resulting command.
+     * @throws SpotException  If input string is invalid.
+     */
+    public static Command parseCommand(String input) throws SpotException {
+        if (input.startsWith("list tasks on")) {
+            if (input.length() <= 14) {
+                throw new SpotException("Spot thinks you might've "
+                        + "forgotten to add a date!");
+            }
+            String d = input.substring(14);
+            LocalDate date = Parser.parseDate(d);
+            return new ListTasksOnCommand(date);
+        } else if (input.startsWith("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("find")) {
+            if (input.length() <= 5) {
+                throw new SpotException("Spot needs more details than that!");
+            }
+            String keyword = input.substring(5);
+            return new FindCommand(keyword);
+        } else if (input.startsWith("mark")) {
+            if (input.length() <= 5) {
+                throw new SpotException("Spot needs more details than that!");
+            }
+            int position = Integer.parseInt(input.substring(5));
+            return new MarkCommand(position);
+        } else if (input.startsWith("unmark")) {
+            if (input.length() <= 7) {
+                throw new SpotException("Spot needs more details than that!");
+            }
+            int position = Integer.parseInt(input.substring(7));
+            return new UnmarkCommand(position);
+        } else if (input.startsWith("delete")) {
+            if (input.length() <= 7) {
+                throw new SpotException("Spot needs more details than that!");
```

Comment:
Consider constants for repeated strings. For instance, the error message "Spot needs more details than that!" appears multiple times. Can help with consistency, just a nit.